### PR TITLE
fix for unit tests: do not call NewAccessWitness in NewEVMTxContext

### PR DESCRIPTION
**Pull request by gballet**
Ensure that the tests pass if I deactivate witnesses in general tests. This includes part of the recommended fix for #47.

:warning: :skull_and_crossbones:  this doesn't change the fact that calculating an address is very slow with the proposed "pedersen hash" scheme.

### CODE REVIEW

**Comment by gballet** [comment]
the test keeps failing because of the following issue:

```
goroutine 79 [runnable]:
github.com/crate-crypto/go-ipa/bandersnatch/fp.mul(0xc0020d8490, 0xc0020d83b0, 0xc0020d8370)
	/home/gballet/go/pkg/mod/github.com/crate-crypto/go-ipa@v0.0.0-20211223165939-ab3f49447206/bandersnatch/fp/element_mul_amd64.s:46 +0x3d7
github.com/crate-crypto/go-ipa/bandersnatch/fp.(*Element).Mul(...)
	/home/gballet/go/pkg/mod/github.com/crate-crypto/go-ipa@v0.0.0-20211223165939-ab3f49447206/bandersnatch/fp/element.go:288
github.com/crate-crypto/go-ipa/bandersnatch.(*PointProj).Double(0xc0020d86d0, 0xc0020d86d0)
	/home/gballet/go/pkg/mod/github.com/crate-crypto/go-ipa@v0.0.0-20211223165939-ab3f49447206/bandersnatch/point.go:305 +0x3c9
github.com/crate-crypto/go-ipa/bandersnatch.PointAffine.IsInPrimeSubgroup({{0xa93a1b95e405524, 0xcbdcd03ccccd3bfe, 0xdd2f6874a5943d08, 0x25e667f67bbea689}, {0x9a5e23d862b9673f, 0x508db07c65c463cd, 0xc2d9b667ccc51dac, 0x6b98f119f248d762}})
	/home/gballet/go/pkg/mod/github.com/crate-crypto/go-ipa@v0.0.0-20211223165939-ab3f49447206/bandersnatch/point.go:373 +0x17a
github.com/crate-crypto/go-ipa/ipa.GenerateRandomPoints(0x100)
	/home/gballet/go/pkg/mod/github.com/crate-crypto/go-ipa@v0.0.0-20211223165939-ab3f49447206/ipa/config.go:204 +0x2de
github.com/crate-crypto/go-ipa/ipa.NewIPASettings()
	/home/gballet/go/pkg/mod/github.com/crate-crypto/go-ipa@v0.0.0-20211223165939-ab3f49447206/ipa/config.go:34 +0x3f
github.com/ethereum/go-ethereum/trie/utils.GetTreeKey({0xc0020d8e8c, 0x14, 0x14}, 0xc00011b320, 0x1)
	/usr/home/gballet/src/go-ethereum/trie/utils/verkle.go:58 +0x2a5
github.com/ethereum/go-ethereum/trie/utils.GetTreeKeyBalance(...)
	/usr/home/gballet/src/go-ethereum/trie/utils/verkle.go:73
github.com/ethereum/go-ethereum/consensus/ethash.accumulateRewards(0xc0001b2180, 0xc000034700, 0xc001f7eb40, {0x131f208, 0x0, 0x0})
	/usr/home/gballet/src/go-ethereum/consensus/ethash/consensus.go:671 +0x671
github.com/ethereum/go-ethereum/consensus/ethash.(*Ethash).Finalize(0xc000234a80, {0x5239b8, 0xc000844000}, 0xc001f7eb40, 0xc000034700, {0x131f208, 0x0, 0x0}, {0x131f208, 0x0, ...})
	/usr/home/gballet/src/go-ethereum/consensus/ethash/consensus.go:597 +0x85
github.com/ethereum/go-ethereum/consensus/beacon.(*Beacon).Finalize(0xc000072e80, {0x5239b8, 0xc000844000}, 0xc001f7eb40, 0xc000034700, {0x131f208, 0x0, 0x0}, {0x131f208, 0x0, ...})
	/usr/home/gballet/src/go-ethereum/consensus/beacon/consensus.go:275 +0x13f
github.com/ethereum/go-ethereum/core.(*StateProcessor).Process(0xc000814c60, 0xc00078f170, 0xc000034700, {0x0, {0x0, 0x0}, 0x0, 0x0, 0x0, {0x0, ...}, ...})
	/usr/home/gballet/src/go-ethereum/core/state_processor.go:90 +0x102f
github.com/ethereum/go-ethereum/core.(*BlockChain).insertChain(0xc000844000, {0xc0007a8008, 0x3e7, 0x3ff}, 0x1, 0x1)
	/usr/home/gballet/src/go-ethereum/core/blockchain.go:1602 +0x21ac
github.com/ethereum/go-ethereum/core.(*BlockChain).InsertChain(0xc000844000, {0xc0007a8008, 0x3e7, 0x3ff})
	/usr/home/gballet/src/go-ethereum/core/blockchain.go:1368 +0xcf2
github.com/ethereum/go-ethereum/cmd/devp2p/internal/ethtest.setupGeth(0xc0003569c0)
	/usr/home/gballet/src/go-ethereum/cmd/devp2p/internal/ethtest/suite_test.go:105 +0x1a9
github.com/ethereum/go-ethereum/cmd/devp2p/internal/ethtest.runGeth()
	/usr/home/gballet/src/go-ethereum/cmd/devp2p/internal/ethtest/suite_test.go:72 +0x94
github.com/ethereum/go-ethereum/cmd/devp2p/internal/ethtest.TestEthSuite(0xc000502340)
	/usr/home/gballet/src/go-ethereum/cmd/devp2p/internal/ethtest/suite_test.go:38 +0x46
testing.tRunner(0xc000502340, 0x4297d8)
	/usr/local/go/src/testing/testing.go:1259 +0x102
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1306 +0x3ab
```

**Comment by gballet** [comment]
Another test fails for the same reasons:

```
goroutine 28 [runnable]:
github.com/crate-crypto/go-ipa/bandersnatch/fp.mul(0xc0001367b0, 0xc000136b30, 0xc000136b30)
	/home/gballet/go/pkg/mod/github.com/crate-crypto/go-ipa@v0.0.0-20211223165939-ab3f49447206/bandersnatch/fp/element_mul_amd64.s:46 +0x3d7
github.com/crate-crypto/go-ipa/bandersnatch/fp.(*Element).Square(...)
	/home/gballet/go/pkg/mod/github.com/crate-crypto/go-ipa@v0.0.0-20211223165939-ab3f49447206/bandersnatch/fp/element.go:295
github.com/crate-crypto/go-ipa/bandersnatch.(*PointProj).Double(0xc000136af0, 0xc000136af0)
	/home/gballet/go/pkg/mod/github.com/crate-crypto/go-ipa@v0.0.0-20211223165939-ab3f49447206/bandersnatch/point.go:298 +0x28c
github.com/crate-crypto/go-ipa/bandersnatch.PointAffine.IsInPrimeSubgroup({{0xa93994b5e405d92, 0xb4796186bef8bb90, 0xaeca7c35d7df39f0, 0x633ee6ff4954bd29}, {0xae71b521ddf23201, 0x6500d1ae2c0eb2d8, 0x6e1942871f7c3c16, 0x4b29a1c92f261a25}})
	/home/gballet/go/pkg/mod/github.com/crate-crypto/go-ipa@v0.0.0-20211223165939-ab3f49447206/bandersnatch/point.go:373 +0x17a
github.com/crate-crypto/go-ipa/ipa.GenerateRandomPoints(0x100)
	/home/gballet/go/pkg/mod/github.com/crate-crypto/go-ipa@v0.0.0-20211223165939-ab3f49447206/ipa/config.go:204 +0x2de
github.com/crate-crypto/go-ipa/ipa.NewIPASettings()
	/home/gballet/go/pkg/mod/github.com/crate-crypto/go-ipa@v0.0.0-20211223165939-ab3f49447206/ipa/config.go:34 +0x3f
github.com/ethereum/go-ethereum/trie/utils.GetTreeKey({0xc000137318, 0x14, 0x14}, 0xc000137250, 0x8)
	/usr/home/gballet/src/go-ethereum/trie/utils/verkle.go:58 +0x2a5
github.com/ethereum/go-ethereum/trie/utils.GetTreeKeyStorageSlot({0xc000137318, 0x14, 0x14}, 0xc000260100)
	/usr/home/gballet/src/go-ethereum/trie/utils/verkle.go:116 +0x197
github.com/ethereum/go-ethereum/core/vm.opSload(0xc0001da4c8, 0xc00039c000, 0xc000582060)
	/usr/home/gballet/src/go-ethereum/core/vm/instructions.go:581 +0x25c
github.com/ethereum/go-ethereum/core/vm.(*EVMInterpreter).Run(0xc00039c000, 0xc000396e40, {0x0, 0x0, 0x0}, 0x0)
	/usr/home/gballet/src/go-ethereum/core/vm/interpreter.go:308 +0xdd9
github.com/ethereum/go-ethereum/core/vm.(*EVM).create(0xc0000ec000, {0x3a5760, 0xc000262360}, 0xc0001379a0, 0x9a91a, 0xc0001e0660, {0xcc, 0x2c, 0xd, 0x8, ...}, ...)
	/usr/home/gballet/src/go-ethereum/core/vm/evm.go:488 +0x928
github.com/ethereum/go-ethereum/core/vm.(*EVM).Create(0xc0000ec000, {0x3a5760, 0xc000262360}, {0xc000094000, 0x970, 0x1300}, 0x9a91a, 0xc0001e0660)
	/usr/home/gballet/src/go-ethereum/core/vm/evm.go:536 +0x212
github.com/ethereum/go-ethereum/core.(*StateTransition).TransitionDb(0xc0000e0500)
	/usr/home/gballet/src/go-ethereum/core/state_transition.go:344 +0x12c6
github.com/ethereum/go-ethereum/accounts/abi/bind/backends.(*SimulatedBackend).callContract(0xc0001dfa00, {0x3af570, 0xc000026098}, {{0xf, 0xfe, 0x93, 0x58, 0xde, 0xdf, 0x26, ...}, ...}, ...)
	/usr/home/gballet/src/go-ethereum/accounts/abi/bind/backends/simulated.go:638 +0x733
github.com/ethereum/go-ethereum/accounts/abi/bind/backends.(*SimulatedBackend).EstimateGas.func1(0xb0112)
	/usr/home/gballet/src/go-ethereum/accounts/abi/bind/backends/simulated.go:536 +0x1d1
github.com/ethereum/go-ethereum/accounts/abi/bind/backends.(*SimulatedBackend).EstimateGas(0xc0001dfa00, {0x3af570, 0xc000026098}, {{0xf, 0xfe, 0x93, 0x58, 0xde, 0xdf, 0x26, ...}, ...})
	/usr/home/gballet/src/go-ethereum/accounts/abi/bind/backends/simulated.go:550 +0x790
github.com/ethereum/go-ethereum/accounts/abi/bind.(*BoundContract).estimateGasLimit(0xc000116f00, 0xc0000cc150, 0x0, {0xc000094000, 0x970, 0x1300}, 0x0, 0xc0001e0680, 0xc0001e06a0, 0xc0001e0660)
	/usr/home/gballet/src/go-ethereum/accounts/abi/bind/base.go:347 +0x29d
github.com/ethereum/go-ethereum/accounts/abi/bind.(*BoundContract).createDynamicTx(0xc000116f00, 0xc0000cc150, 0x0, {0xc000094000, 0x970, 0x1300}, 0xc0000d3b00)
	/usr/home/gballet/src/go-ethereum/accounts/abi/bind/base.go:264 +0x285
github.com/ethereum/go-ethereum/accounts/abi/bind.(*BoundContract).transact(0xc000116f00, 0xc0000cc150, 0x0, {0xc000094000, 0x970, 0x1300})
	/usr/home/gballet/src/go-ethereum/accounts/abi/bind/base.go:376 +0x1c7
github.com/ethereum/go-ethereum/accounts/abi/bind.DeployContract(0xc0000cc150, {{{0x0, 0x0}, {0x0, 0x0}, 0x0, {0xc0001db2a0, 0xa}, 0x0, 0x0, ...}, ...}, ...)
	/usr/home/gballet/src/go-ethereum/accounts/abi/bind/base.go:137 +0x46b
github.com/ethereum/go-ethereum/contracts/checkpointoracle/contract.DeployCheckpointOracle(0xc0000cc150, {0x3b8408, 0xc0001dfa00}, {0xc00053adc0, 0x3, 0x3}, 0xc0001e0100, 0xc0001e0120, 0xc0001e0460)
	/usr/home/gballet/src/go-ethereum/contracts/checkpointoracle/contract/oracle.go:49 +0x310
github.com/ethereum/go-ethereum/contracts/checkpointoracle.TestCheckpointRegister(0xc000202b60)
	/usr/home/gballet/src/go-ethereum/contracts/checkpointoracle/oracle_test.go:190 +0x7bb
testing.tRunner(0xc000202b60, 0x310ce8)
	/usr/local/go/src/testing/testing.go:1259 +0x102
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1306 +0x3ab
```

**Comment by gballet** [comment]
@jwasinger this PR is meant as an inspiration for an update of #47, in which the function signatures don't get changed.

I realize that some of these changes are already addressed by another one of your PRs, I just wanted to see the tests pass.

File: consensus/ethash/consensus.go [hunk 1]
```diff
@@ -661,14 +661,20 @@ func accumulateRewards(config *params.ChainConfig, state *state.StateDB, header
...
 		reward.Add(reward, r)
 	}
-	coinbase := utils.GetTreeKeyBalance(header.Coinbase.Bytes())
-	state.Witness().TouchAddress(coinbase, state.GetBalance(header.Coinbase).Bytes())
+	if config.IsCancun(header.Number) {
+		coinbase := utils.GetTreeKeyBalance(header.Coinbase.Bytes())
+		if state.Witness() != nil {
+			state.Witness().TouchAddress(coinbase, state.GetBalance(header.Coinbase).Bytes())
+		}
+	}
```
Comment (jwasinger) [inline consensus/ethash/consensus.go:678]: can be:
```
	if state.Witnes() != nil {
		coinbase := utils.GetTreeKeyBalance(header.Coinbase.Bytes())
	        state.Witness().TouchAddress(coinbase, state.GetBalance(header.Coinbase).Bytes())
	}
```

**Comment by jwasinger** [comment]
lgtm

**Pull request merged**